File: 890-find-and-replace-pattern/890-find-and-replace-pattern.js
/**
 * @param {string[]} words
 * @param {string} pattern
 * @return {string[]}
 */
var findAndReplacePattern = function(words, pattern) {
    var res = [];
    for(let i = 0;i<words.length;i++)
    {
        var flag = 1;
        var mp = new Map();
        var str = words[i];
        if(str.length != pattern.length)
        {
            continue;
        }
        for(let k = 0;k<str.length;k++)
        {
            if(mp.has(pattern[k]))
            {
                if(mp.get(pattern[k])!=str[k])
                {
                    flag = 0;
                    break;
                }
            }
            else
            {
                mp.set(pattern[k],str[k]);
            }
        }
        mp.clear();
        for(let k = 0;k<str.length;k++)
        {
            if(mp.has(str[k]))
            {
                if(mp.get(str[k])!=pattern[k])
                {
                    flag = 0;
                    break;
                }
            }
            else
            {
                mp.set(str[k],pattern[k]);
            }
        }
        if(flag)
            res.push(words[i]);
    }
    return res;
};

File: 890-find-and-replace-pattern/890-find-and-replace-pattern.cpp
class Solution {
public:
    vector<string> findAndReplacePattern(vector<string>& words, string pattern) {
        vector<string>res;
        
        for(auto it : words){
            unordered_map<char,char>mp;
            unordered_set<char>st;
            int i;
            for(i=0;i<it.size();i++){
                if(mp.find(pattern[i])!=mp.end()){
                    if(mp[pattern[i]]==it[i]) continue;
                    else break;
                }
                else{
                    
                    if(st.find(it[i])!=st.end()) break;
                    
                    mp[pattern[i]] = it[i];
                    st.insert(it[i]);
                }
            }
            if(i==it.size()) res.push_back(it);
        }
        return res;
    }
};